detox/src/configuration/composeAppsConfig.js

```diff
@@ -137,16 +144,31 @@ function validateAppConfig({ appConfig, appPath, deviceConfig, errorComposer })
     });
   }
 
-  if (allowedAppTypes && !appConfig.binaryPath) {
-    throw errorComposer.missingAppBinaryPath(appPath);
-  }
+  if (appConfig.type !== 'android.cloud') {
+    if (allowedAppTypes && !appConfig.binaryPath) {
+      throw errorComposer.missingAppBinaryPath(appPath);
+    }
 
-  if (appConfig.launchArgs && !_.isObject(appConfig.launchArgs)) {
-    throw errorComposer.malformedAppLaunchArgs(appPath);
+    if (appConfig.launchArgs && !_.isObject(appConfig.launchArgs)) {
+      throw errorComposer.malformedAppLaunchArgs(appPath);
+    }
+
+    if (appConfig.type !== 'android.apk' && appConfig.reversePorts) {
+      throw errorComposer.unsupportedReversePorts(appPath);
+    }
   }
+  else {
+    if (!_.isString(appConfig.app)) {
+      throw errorComposer.invalidCloudAppUrl(appPath);
+    }
 
-  if (appConfig.type !== 'android.apk' && appConfig.reversePorts) {
-    throw errorComposer.unsupportedReversePorts(appPath);
+    if (!_.isString(appConfig.appClient)) {
+      throw errorComposer.invalidCloudAppClientUrl(appPath);
+    }
+  }
+  const ignoredCloudConfigParams = _.difference(Object.keys(appConfig), supportedCloudAppsConfig);
+  if (configurationName === 'android.cloud.release' && ignoredCloudConfigParams.length > 0 ) {
+    logger.warn(`[AppConfig] The properties ${ignoredCloudConfigParams.join(', ')} are not honoured for device type 'android.cloud'`); 
```

Comment:
**[NOT A MUST HAVE]** **[UX]**
Some params may not apply to particular device types. For example, recipeUUID will not apply to `android.attached` device type.

This may be confusing for the users, so detox should have a way to inform users about ignored params. We have implemented this for `android.cloud` device type for now.